src/pydvl/valuation/samplers/classwise.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import annotations
+
+from typing import Callable, Generator
+
+import numpy as np
+from numpy.typing import NDArray
+
+from pydvl.valuation.dataset import Dataset
+from pydvl.valuation.samplers.base import EvaluationStrategy, IndexSampler
+from pydvl.valuation.samplers.powerset import PowersetSampler
+from pydvl.valuation.samplers.utils import StochasticSamplerMixin
+from pydvl.valuation.types import CSSample, IndexSetT, SampleGenerator
+from pydvl.valuation.utility.base import UtilityBase
+
+__all__ = ["ClasswiseSampler", "get_unique_labels"]
+
+
+def get_unique_labels(array: NDArray) -> NDArray:
+    """Labels of the dataset."""
+    # Object, String, Unicode, Unsigned integer, Signed integer, boolean
+    if array.dtype.kind in "OSUiub":
+        return np.unique(array)
+    raise ValueError("Dataset must be categorical to have unique labels.")
```

Comment:
This message is a bit confusing, because there is no categorical dtype in numpy. Maybe something like:
```suggestion
raise ValueError(
        f"Input array has an unsupported data type for categorical labels: {array.dtype}. "
        "Expected types: Object, String, Unicode, Unsigned integer, Signed integer, or Boolean."
    )
```